### ISSUE TITLE
Mock the commands

### ISSUE THREAD

**Issue by gre** [LedgerHQ/ledger-live-desktop#1251]
The commands are the way to execute some job on the internal process from the renderer.

this is quite simple to do, we basically need a way to "handle the command execution" for a given command at runtime and somehow "unhandle" when no longer needed..

## The usecase

**The usecase is to write tests.**

We have not chosen any framework yet, but whatever framework use, what's sure is we need a way to mock data at some part of the app. I don't think it's easily feasible to mock all HTTP calls and all APDU calls to the device because this is **highly unpredictable** and not really interesting to go that deep. However what's easy to do is to much commands, which nicely happen to be all things related to device and even to blockchains interactions (except for some part of ETH today, but this can be improved).

In the scope of such usecase, there is also #1228 that will need to be worked on, as well as a more generic way to expose this as a runtime api.

---

So basically,

we want to allow to override the default execution of a command called from the renderer to typically allow our QA folks to implement tests and shortcut things like "getIsGenuine" or even mock the "listDevices",..

## code sample

typically we would allow scripts like (pseudocode):

```js
const customGetIsGenuine = () => Observable.create(o => { o.next("9000") })

const commandHandler = ipc.handleCommand("getIsGenuine", customGetIsGenuine)

await navigationTo("onboarding")
await nextStep()
await nextStep()
await checkOn(".pinCodeCheckbox")
await checkOn(".WritenMyOwnSeed")
await clickOn(".checkOnboarding")
await expectCommandToHappen(commandHandler)

commandHandler.unhandle();

assert("everything is ok")
```

**Comment by meriadec** [comment]
:100: 

Started a POC on integrating Spectron (end-to-end utility to launch tests on builded electron app), [see here](https://github.com/meriadec/ledger-live-desktop/commit/d71a857cd80d9a09796b93b48db9ac8eca5b991c). I think we can totally build something around to easily write integration test (using id/class convention to select elements, etc.)

**Comment by gre** [comment]
totally.

in my mind, for the functional tests we can go with mocking the commands and the bridge.
and for possible unit tests, we can test each commands / bridge individually.
this would be a full coverage and so functional don't have to worry about the unit tests part.

**Comment by gre** [comment]
i'm not sure if mocking the commands is ultimately the best level where mock will happen.
Ultimately it could instead be done at the Transport level. it would be lower level but everything is possible (just need to implement apdu protocol \o/) and we can in future, replace by actual simulators.
